contracts/core/Registry.sol

```diff
@@ -21,27 +36,75 @@ contract Registry {
     // attestationAddr = addr(hash(id, name, owner))
     // OR deploy right away with CREATE2 as proxy so they can do anything (or just forward ERC20s and ETH)
     // id => IdentityDetails
-    function createIdentity(IdentityDetails memory _identityDetails, address[] memory _owners) external returns (uint256) {
 
+    // ASK: changing the function signature as attestationAddr has to be generated from the name and caller
+    /// @notice Creates a new identity
+    /// @dev This will also set the attestation address generated from msg.sender and name
+    function createIdentity(
+        string memory name,
+        Metadata memory metadata,
+        address[] memory _owners
+    ) external returns (uint256) {
+
+        IdentityDetails memory identityDetails = IdentityDetails(
+            name,
+            metadata,
+            // ASK : should identityId be used as the hash for attestationAddr?
+            // cause project can't have same attestAddr across chains
+            generateAttestationAddr(name, msg.sender)
```

Comment:
@zobront Do we need the counter to generate the attestation address  ? Check comment above

Reply:
I don't think `msg.sender` should be a part of this. We want the attestation address to be consistent as long as the name doesn't change. 

I think `identityId` (immutable) and `name` is the right thing.

I don't think it's a problem to have different attestation addresses across chains, because we'll need to be attesting separately in every chain anyways.

Reply:
but having a different attestation address for the same owner and name would mean that identity would have to rebuild attestations across chains. I was thinking of a scenario where attestations could be ported by a 3rd part.y from one chain to another

Reply:
Yea, if it's possible, it would be great to give people the same address on different chains. I think we would have to use `msg.sender` in order to do that?

Reply:
### Overview
Since the `identityId` changes on each new identity created the address would be different on each chain, we cannot guarantee what the `identityId` would be with the current code (see below):

```javascript
function _generateAttestationAddr(uint _identityId, string memory _name) internal pure returns (address) {
    bytes32 attestationHash = keccak256(
        abi.encodePacked(_identityId, _name)
    );

    return address(uint160(uint256(attestationHash)));
 }
```

### Suggested Changes

Adding in the `sender` and removing the `identityId` would allow us to make sure the address is the same on each chain. I don't think we would ever have a collision where the `_name` and `_sender` are the same within a single `Identity` 🤔 

```javascript
function _generateAttestationAddr(
    string memory _name,
    address _sender
) internal pure returns (address) {
    bytes32 attestationHash = keccak256(abi.encodePacked(_name, _sender));

    return address(uint160(uint256(attestationHash)));
}
```

Let me know what you all think 🙌 There may be something I am missing 🤔

cc: @zobront @0xZakk @thelostone-mc @KurtMerbeth

Reply:
This seems to be the last big open question. I'm not sure either of the solutions above is right.

## CONSTRAINTS 

Here are the constraints we're working with...
1) make it consistent across chains
2) make it unique
3) have it stay unchanged when important values (like name) are unchanged, but reset when those values are changed


## PROBLEMS 

But the solutions above all miss some of these...
- if we include a globally incremented identityId, we can't be consistent across chains
- if we don't include identityId, we aren't necessarily unique (for example, current implementation would let the same user create two identities with the same name, and they'd have the same address)

Additionally, there is the slightly awkward problem that the attestation address is based on the original msg.sender, but that isn't saved, so when we update it we use the new msg.sender ([see here](https://github.com/allo-protocol/allo-v2/blob/da3cf15f6c4f10a85d9bff81319109ac4ddcde1a/contracts/core/Registry.sol#L108). No bueno.

## SOLUTION

The only solution I can think of that solves for all constraints is...
- when an identity is created, we immediately create an `identityId` that is the hash of the `msg.sender` and a `nonce` that is incremented per `msg.sender` (ie it's stored in a mapping and incremented each time they create an identity)
- that `identityId` is saved as a part of the `identity` struct
- the `attestationAddress` (or whatever we choose to call it) is the hash of `identityId` and `name`
- we can NEVER change `identityId`, it's immutable once the identity is created and has nothing to do with ownership, the original deployer has no ownership or rights
- but if we change name, we just use the old `identityId` to generate a new `attestationAddress`

I think this solves for all constraints. Let me know if it's unclear or if we should chat about it more.

Reply:
This means :

- If owner updates name -> use same identityId and generate new attestationAddress
- If owners are updated via grantRoles -> we just revert them and ask them to create a new identity themselves

This feels fair. The only downside is that we can't create a new identity for them cause we won't be able to port their owners and members cause those can be iterated through. That feels like a big issue.

My question: 
Why do we need to create a new identity / generate a new attention address anytime the owner is changed. 
I would say grantRole / revokeRole shouldn't change things in your identity

Reply:
Thinking about a little further, even having a identityId as a counter makes sense  ?
A project with identityId 1 on chain A will be different from identityId 1 on chain B 
We could instead simplify this by saying :
- name cannot be updated 
-  The attestation address/. anchor is generated from `msg.sender`, `owner` and `name`
- And all lookups are by anchor / attestation address.

You can change members without any issues.
If you want to change owners / name -> you would have to create a new identity 

@0xZakk + @nfrgosselin

Reply:
Per our chat on standup, I'm dropping a list of use cases we want to consider for the identity registry.

**most important things**
- allowing projects to be _created_ easily
- allowing projects to build up reputation
- preventing fake projects / impersonating another project (e.g. creating a fake Gitcoin)
- preventing partial owner(s) of a group-owned identity stealing it from the group without consent

**things we want to be possible (even if they have some friction)**
- rebranding a project
- transferring ownership of a project

**things we want to limit**
- creating / selling false identities

Reply:
Okie marking this as resolved as the solution we came with @zobront  on Friday covers our usecases